beets/util/__init__.py

```diff
@@ -1108,3 +1108,31 @@ def wrapper(self):
         return value
 
     return wrapper
+
+
+def expand_key_globs(included_keys, all_keys_cb):
+    """Expand '*' placeholders in a list of key names.
+
+    The list of all possible keys is obtained by calling `all_keys_cb()`.
+    This is only called if `included_keys` contains globs.
+
+    """
+    glob_keys = set(k for k in included_keys if '*' in k)
+    keys = set(k for k in included_keys if k not in glob_keys)
+
+    if glob_keys:
+        all_keys = all_keys_cb()
+
+        matchers = []
+        for key in glob_keys:
+            pattern = re.escape(key).replace(r'\*', '.*') + '$'
+            matchers.append(re.compile(pattern))
```

Comment:
You may have considered this already, but could the [fnmatch.fnmatch](https://docs.python.org/3/library/fnmatch.html#fnmatch.fnmatch) function be useful for this to avoid constructing regexes explicitly? Or else [fnmatch.translate](https://docs.python.org/3/library/fnmatch.html#fnmatch.translate) if we still want to use proper regexes?

Reply:
fnmatch implements a slightly more complex syntax (`?` wildcards, `[abc]` character sequences). Is that what we want?

Reply:
Yeah, good point. While they seem cool, I can't imagine anyone actually wanting to use them. So maybe it's not worth it.